source/lac/trilinos_sparse_matrix.cc

```diff
@@ -126,9 +126,6 @@ namespace TrilinosWrappers
           return;
         }
 
-      // otherwise first flush Trilinos caches
-      matrix->compress ();
-
```

Comment:
I'm going to add another test `mpi/trilinos_sparse_matrix_03` that verifies the semantics one would expect in this context. It actually points to another problem, namely that one can call `begin(row)` on a locally owned row, but not `end(row)` if `row` is the last row the current processor owns -- awkward...